### PR TITLE
[EDMTP-303] Handlers processing unification and cleanup

### PR DESCRIPTION
**Pull request by smiakchilo**

### CODE REVIEW

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/target/TargetImpl.java [hunk 1]
```diff
@@ -513,12 +509,22 @@ private static void populateAnnotationProperty(Annotation context, Method method
...
     }
 
+    private static String getPropertyPrefix(Annotation annotation) {
+        String result = StringUtils.EMPTY;
+        if (annotation.annotationType().isAnnotationPresent(MapProperties.class)) {
+            result = annotation.annotationType().getDeclaredAnnotation(MapProperties.class).prefix();
+        } else if (annotation.annotationType().isAnnotationPresent(PropertyMapping.class)) {
+            result = annotation.annotationType().getDeclaredAnnotation(PropertyMapping.class).prefix();
+        }
+        return result;
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/target/TargetImpl.java:525]: (сosmetic) you can use returns at lines 521 and 523 and avoid declaring a string

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+public class ClassUtil {
+
+    /**
+     * Default (hiding) constructor
+     */
+    private ClassUtil() {
+    }
+
+    /**
+     * Retrieves a sequential list of {@link Source} objects representing manageable members that belong
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java:50]: (cosmetic) belong to

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+     * a certain {@code Class} (and its superclasses) and match specific criteria
+     * @param targetClass The class to extract sources from
+     * @return List of {@code Source} objects
+     */
+    public static List<Source> getSources(Class<?> targetClass) {
+        return getSources(targetClass, Collections.emptyList());
+    }
+
+    /**
+     * Retrieves a sequential list of {@link Source} objects representing manageable members that belong
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java:60]: (same here)

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+import com.exadel.aem.toolkit.plugin.util.ordering.OrderingUtil;
+
+/**
+ * Contains utility methods for parsing AEM components' classes and extracting information related to UI rendering
+ */
+public class ClassUtil {
+
+    /**
+     * Default (hiding) constructor
+     */
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java:45]: (cosmetic) perhaps the constructor is hidden, not hiding?)

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    private ClassUtil() {
+    }
+
+    /**
+     * Retrieves a sequential list of {@link Source} objects representing manageable members that belong
+     * a certain {@code Class} (and its superclasses) and match specific criteria
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java:51]: `match specific criteria` - perhaps we can clarify , which criteria?

Comment (smiakchilo) [reply aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java:51]: Fixed

File: aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java (hunk 3 above)
Comment (smiakchilo) [reply aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java:60]: Fixed

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/meta/MapProperties.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+ */
+@Target(ElementType.ANNOTATION_TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface MapProperties {
+
+    /**
+     * Defines mapping rules for the properties (methods) of the current annotation. Default rule is that all the property
+     * values of an appropriate type (string, number, enum) are mapped. The specific rules are:
+     * <p>1) It one or more names is specified, only these properties are mapped.</p>
+     * <p>2) If one or more names specified with the prepended negation sign ("{@code !}"), all the properties <i>save for</i>
```
[the remark this answers is not in the document]
Comment (smiakchilo) [reply aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/meta/MapProperties.java:33]: This is fixed in the ongoing PR, thanks

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/datepicker/DatePicker.java [hunk 7]
```diff
@@ -98,7 +97,6 @@
      * default format otherwise
      * @return Value equal to "String", or none
      */
-    @IgnorePropertyMapping
     TypeHint typeHint() default TypeHint.NONE;
```
[the remark this answers is not in the document]
Comment (smiakchilo) [reply aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/datepicker/DatePicker.java:100]: Fixed

**Review by liubou-masiuk — APPROVED** [review]

**Pull request merged**